From: Marijn Kruisselbrink <mkruisselbrink@kde.org>
Date: Mon, 01 Aug 2011 16:02:21 +0000
Subject: Fix bug introduced with svn revision 1023186 (git f62946276067d0a4b28e2039cfb506d7e8d678e4).
X-Git-Url: http://quickgit.kde.org/?p=kdelibs.git&amp;a=commitdiff&amp;h=d7daf264b9e8d2c98f210f2717c375026d585481
---
Fix bug introduced with svn revision 1023186 (git f62946276067d0a4b28e2039cfb506d7e8d678e4).
Don't produce an infinite loop when trying to get an invalid unit.
CCMAIL: petri.damsten@kdemail.net
---


--- a/kunitconversion/converter.cpp
+++ b/kunitconversion/converter.cpp
@@ -165,7 +165,7 @@ UnitPtr Converter::unit(int unitId) cons
             return unitClass;
         }
     }
-    return unit(InvalidUnit);
+    return category[InvalidCategory]->defaultUnit();
 }
 
 UnitCategory* Converter::category(const QString& category) const


